### PR TITLE
RFQ VIP support + fixes

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
- Fix missing call handlers in manifest.
- Support RFQ VIP
- Misc schema cleanup to hopefully speed up indexing.

:ballot_box_with_check:  already deployed on Pending

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -15,8 +17,9 @@ type Transaction @entity {
     blockNumber: BigInt!
     timestamp: BigInt!
     fills: [Fill!]!
-    nativeOrderFills: [NativeOrderFill!]!
-    swaps: [Swap!]!
+    nativeOrderFills: [NativeOrderFill!]! @derivedFrom(field: "transaction")
+    swaps: [Swap!]! @derivedFrom(field: "transaction")
```
Comment (xianny) [inline schema.graphql:21]: @derivedFrom is cool, didn't know about it!

Comment (xianny) [reply schema.graphql:21]: Did you find it in docs somewhere? I can't find it anywhere on the Internet except a random Medium article, wondering what other decorators I'm missing.

File: src/mappings/exchange-proxy.ts [hunk 2]
```diff
@@ -32,7 +36,7 @@ export function handleTransformedERC20Event(event: TransformedERC20): void {
     inputToken.save();
     outputToken.save();
 
-    let swap = new Swap(tx.id + '-' + BigInt.fromI32(tx.swaps.length).toString());
+    let swap = new Swap(tx.id + '-' + event.logIndex.toString());
```
Comment (xianny) [inline src/mappings/exchange-proxy.ts:39]: I like this

File: src/mappings/exchange-proxy.ts [hunk 3]
```diff
@@ -302,17 +286,92 @@ export function handleSellToUniswapCall(call: SellToUniswapCall): void {
...
+
+    let tx = transactionFindOrCreate(call.transaction.hash, call.block);
+    let fills = findSwapEventFills(tx); // sus
+    inputToken.swapVolume = inputToken.swapVolume.plus(fillData.sellAmount);
+    outputToken.swapVolume = outputToken.swapVolume.plus(call.outputs.outputTokenAmount);
+    inputToken.save();
+    outputToken.save();
+
+    let r = getRandomNumber();
+    let swap = new Swap(tx.id + '-' + r.toString());
```
Comment (xianny) [inline src/mappings/exchange-proxy.ts:314]: no logIndex? :(

Comment (xianny) [reply src/mappings/exchange-proxy.ts:314]: Can we make it less random, like `id + call.from` maybe? Thinking about possible query patterns.

File: src/mappings/exchange-proxy.ts [hunk 4]
```diff
@@ -407,10 +466,9 @@ function findSellToUniswapEventFills(tx: Transaction, call: SellToUniswapCall):
 }
 
 export function findSwapEventFills(tx: Transaction, logIndex: BigInt | null = null): Fill[] {
```
Comment (xianny) [inline src/mappings/exchange-proxy.ts:468]: I *think* this makes sense...

Looks like when a multi or batch swap happens, we emit MultiHopFillCall/BatchFillCall AND multiple ERC20BridgeTransfer/BridgeFill/LimitOrderFilled/RfqOrderFilled/WhateverFillzz?

And I think this is doing:
```
1. We save all the incoming fills to the tx
2. When we detect a batch or multi call
3. We create a `Swap` abstraction to group the fills with
4. Look up the tx and find the fills that happened at the expected logIndex
```

Is logIndex reliable? Would it be more accurate to use something like taker fill amounts maybe?

File: schema.graphql [hunk 5]
```diff
@@ -15,8 +17,9 @@ type Transaction @entity {
     blockNumber: BigInt!
     timestamp: BigInt!
     fills: [Fill!]!
```
Comment (xianny) [inline schema.graphql:19]: why don't we derive fills too?

File: schema.graphql (hunk 1 above)
Comment (dorothy-zbornak) [reply schema.graphql:21]: Yeah, it's in the tutorial.

File: src/mappings/exchange-proxy.ts (hunk 3 above)
Comment (dorothy-zbornak) [reply src/mappings/exchange-proxy.ts:314]: Calls have no log index and it needs to be unique. All the param inputs can be duplicated multiples times in a tx, so I just went with a random ID. Could do a counter entity but this is probably slightly faster to sync? Unless we really need a monotonically increasing ID here?

File: src/mappings/exchange-proxy.ts (hunk 4 above)
Comment (dorothy-zbornak) [reply src/mappings/exchange-proxy.ts:468]: Yup, you got it.

The fill amounts are pretty unpredictable, even the tokens are unpredictable with multihops.

File: schema.graphql (hunk 5 above)
Comment (dorothy-zbornak) [reply schema.graphql:19]: we can't access derived fields in the mapping script :disappointed:

**Pull request merged**